### DISCUSSION TITLE
Alternatives to spy() for production env?

### DISCUSSION THREAD

**Discussion by nick-delirium** [mobxjs/mobx#3695] [Q&A]
Hi. I'm a maintainer from openreplay, we are making tracking tools to debug web sessions, and, of course, we have a mobx plugin that should (in theory) observe and track state changes and mutations, year ago, during its creation, original creator of that plugin used `spy` simply because its easiest way to achieve global level tracking without much troubles. [Source code here](https://github.com/openreplay/openreplay/blob/main/tracker/tracker-mobx/src/index.ts)

Well of course, it does not really work in production env since `spy` function returns no-op error.

Is there any approach to achieve similar or as close as possible functionality where we can observe mutations? In theory, we can let users attach listeners to every store, but its not the best or desirable approach.

**Comment by nick-delirium** [comment]
ok I guess I'm on my own here 👍

**Comment by mweststrate** [comment]
Mobx-utils.deepObserve or shipping the dev build for mobx are the closets
things
